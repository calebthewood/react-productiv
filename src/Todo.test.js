import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

const initialTestTodo = {
  id: 1,
  title: "Code!",
  description: "Write some code",
  priority: 2,
};

it("renders without crashing", function () {
  render(<Todo todo={initialTestTodo} />);
});

it("snapshot", function () {
  const { container } = render(<Todo todo={initialTestTodo} />);
  expect(container).toMatchSnapshot();
});

it("renders todo", function () {
  const result = render(<Todo todo={initialTestTodo} />);
  expect(result.queryByText("Code!")).toBeInTheDocument();
});

// it("renders todo", function () {
//   const result = render(<Todo todo={initialTestTodo} />);
//   expect(result.queryAllByText("Write some code")).toBeInTheDocument();
// });

it("does not render button", function () {
  const result = render(<Todo todo={initialTestTodo} />);
  expect(result.queryByText("Edit")).not.toBeInTheDocument();
});
