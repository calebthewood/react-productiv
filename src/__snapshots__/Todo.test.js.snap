// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Code!
      </b>
       
      <small>
        2
      </small>
    </div>
    <div>
      <small>
        Write some code
        .
      </small>
    </div>
  </div>
</div>
`;
